### PR TITLE
Rate Limiting docs

### PR DESCRIPTION
**Pull request by alpeb**

### CODE REVIEW

File: linkerd.io/content/2-edge/features/rate-limiting.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
+title: Rate Limiting
+description: Linkerd offers a simple and performant HTTP local rate limiting solution to protect services from misbehaved clients
+---
+
+Rate limiting helps protecting a service by controlling its inbound traffic flow
```
Comment (kflynn) [inline linkerd.io/content/2-edge/features/rate-limiting.md:6]:
```suggestion
Rate limiting helps protect a service by controlling its inbound traffic flow
```

File: linkerd.io/content/2-edge/features/rate-limiting.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+task](../../tasks/configuring-rate-limiting/) for an example guide on deploying
+rate limiting, and the [HTTPLocalRateLimitPolicy reference
+doc](../../reference/rate-limiting/).
+
+## Scope
+
+Linkerd offers a _local_ rate limiting solution, which means that each inbound
+proxy performs the limiting for the pod. This is unlike _global_ rate limiting,
+which takes into account each service's replica to track the global requests
+volume, the trade-off being this requires an external service and is more
```
Comment (kflynn) [inline linkerd.io/content/2-edge/features/rate-limiting.md:20]:
```suggestion
which takes into account all replicas for each service to track  global request
volume. Global rate limiting requires an additional controller to track everything and is thus more
```

File: linkerd.io/content/2-edge/features/rate-limiting.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Fairness
+
+In the `HTTPLocalRateLimitPolicy` CR you can optionally configure a rate limit
+to apply to all the inbound traffic for a given Server, regardless of the
+source.
+
+Additionally, you can specify fairness among clients by declaring a limit per
+identity. This avoids specific clients gobbling all the rate limit quota and
+affect all the other clients. Note that all unmeshed sources (which don't have
```
Comment (kflynn) [inline linkerd.io/content/2-edge/features/rate-limiting.md:31]:
```suggestion
affecting other clients. Note that all unmeshed sources (which don't have
```

File: linkerd.io/content/2-edge/features/rate-limiting.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+(GCRA)](https://en.wikipedia.org/wiki/Generic_cell_rate_algorithm) to implement
+rate limiting, which is more performant than the token bucket and leaky bucket
+algorithms usually used for rate limiting.
+
+The GCRA has two parameters: cell rate and tolerance.
+
+In its virtual scheduling description, the algorithm determines a theoretical
+arrival time, representing the 'ideal' arrival time of a cell (request) if cells
+(requests) were transmitted at equal intervals of time, corresponding to the
+cell rate. How close a flow of requests should abide to that arrival time is
```
Comment (kflynn) [inline linkerd.io/content/2-edge/features/rate-limiting.md:49]:
```suggestion
cell rate. How closely the flow of requests should abide to that arrival time is
```

File: linkerd.io/content/2-edge/features/rate-limiting.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+determined by the tolerance parameter.
+
+In Linkerd we derive the cell rate from the `requestsPerSecond` entries in
+`HTTPLocalRateLimitPolicy` and the tolerance is set to one second. This helps
+accommodating small variations or occasional bursts in traffic while ensuring
```
Comment (kflynn) [inline linkerd.io/content/2-edge/features/rate-limiting.md:54]:
```suggestion
accommodate small variations or occasional bursts in traffic while ensuring
```

File: linkerd.io/content/2-edge/reference/rate-limiting.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+`all-unauthenticated`.
+{{< /note >}}
+
+## HTTPLocalRateLimitPolicy Spec
+
+{{< keyval >}}
+| field| value |
+|------|-------|
+| `targetRef`| A reference to the [Server](../../reference/authorization-policy/#server) this policy applies to. |
+| `total.requestsPerSecond`| Overall rate limit, which all traffic coming to the `targetRef` should abide. If unset no overall limit is applied. |
```
Comment (kflynn) [inline linkerd.io/content/2-edge/reference/rate-limiting.md:26]:
```suggestion
| `total.requestsPerSecond`| Overall rate limit for all traffic sent to the `targetRef`. If unset no overall limit is applied. |
```

File: linkerd.io/content/2-edge/reference/rate-limiting.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+| `total.identity`| Fairness for individual identities; each separate client, grouped by identity, will have this rate limit. The `requestsPerSecond` value should be less than or equal to `total.requestsPerSecond` (if set). |
```
Comment (kflynn) [inline linkerd.io/content/2-edge/reference/rate-limiting.md:27]:
```suggestion
| `identity.requestsPerSecond`| Fairness for individual identities; each separate client, grouped by identity, will have this rate limit. If `total.requestsPerSecond` is also set, `identity.requestsPerSecond` cannot be greater than `total.requestsPerSecond`. |
```

File: linkerd.io/content/2-edge/reference/rate-limiting.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+| `overrides`| An array of [overrides](#overrides) for traffic from specific clients. The `requestsPerSecond` values should be less than or equal to `total.requestsPerSecond` (if set). |
```
Comment (kflynn) [inline linkerd.io/content/2-edge/reference/rate-limiting.md:28]:
```suggestion
| `overrides`| An array of [overrides](#overrides) for traffic from specific client). |
```

File: linkerd.io/content/2-edge/reference/rate-limiting.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+{{< /keyval >}}
+
+### Overrides
+
+{{< keyval >}}
+| field| value |
+|------|-------|
+| `requestsPerSecond`| The number of requests per second allowed from clients matching `clientRefs`. |
```
Comment (kflynn) [inline linkerd.io/content/2-edge/reference/rate-limiting.md:36]:
```suggestion
| `requestsPerSecond`| The number of requests per second allowed from clients matching `clientRefs`.  If `total.requestsPerSecond` is also set, the `requestsPerSecond` for each `overrides` entry cannot be greater than `total.requestsPerSecond`. |
```

File: linkerd.io/content/2-edge/tasks/configuring-rate-limiting.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+  podSelector:
+    matchLabels:
+      app: web-svc
+  port: http
+  proxyProtocol: HTTP/1
+EOF
+```
+
+Now we can apply the `HTTPLocalRateLimitPolicy` resource pointing to that
+`Server`. For now, we'll constraint ourselves to just setting a limit of 4 RPS
```
Comment (kflynn) [inline linkerd.io/content/2-edge/tasks/configuring-rate-limiting.md:66]:
```suggestion
`Server`. For now, we'll just sett a limit of 4 RPS:
```

**Review by kflynn — APPROVED** [review]
Really nice! I've suggested some small changes, but this is really helpful. 🙂

File: linkerd.io/content/2-edge/features/rate-limiting.md (hunk 5 above)
Comment (alpeb) [reply linkerd.io/content/2-edge/features/rate-limiting.md:54]: I clearly have grammar holes on when to use gerunds and when not :laughing:

File: linkerd.io/content/2-edge/reference/rate-limiting.md (hunk 7 above)
Comment (alpeb) [reply linkerd.io/content/2-edge/reference/rate-limiting.md:27]: Good find! :sweat_smile:

**Pull request merged**